src/runtime_src/core/tools/common/ReportPlatforms.cpp

```diff
@@ -0,0 +1,192 @@
+
+  boost::property_tree::ptree pt_clocks;
+  auto raw = xrt_core::device_query<qr::clock_freq_topology_raw>(dev);
+  auto clock_topology = reinterpret_cast<const clock_freq_topology*>(raw.data());
+  for(int i = 0; i < clock_topology->m_count; i++) {
+    boost::property_tree::ptree clock;
+    clock.add("id", clock_topology->m_clock_freq[i].m_name);
+    clock.add("freq_mhz", clock_topology->m_clock_freq[i].m_freq_Mhz);
+    pt_clocks.push_back(std::make_pair("", clock));
+  }
+  pt_platform.put_child("clocks", pt_clocks);
+  
+  auto macs = mac_addresses(dev);
+  if(!macs.empty())
+    pt_platform.put_child("macs", macs);
+  
+  ptree.push_back(std::make_pair("", pt_platform));
+  // There can only be 1 root node
+  pt.add_child("platforms", ptree);
+}
+
+void 
+ReportPlatforms::writeReport( const xrt_core::device * dev,
+                                  const std::vector<std::string> & /*_elementsFilter*/, 
+                                  std::iostream & output) const
+{
+  boost::property_tree::ptree pt;
+  boost::property_tree::ptree empty_ptree;
+  getPropertyTreeInternal(dev, pt);
+
+  output << "Platform\n";
+  boost::property_tree::ptree& platforms = pt.get_child("platforms", empty_ptree);
+  for(auto& kp : platforms) {
+    boost::property_tree::ptree& pt_platform = kp.second;
+    boost::property_tree::ptree& pt_static_region = pt_platform.get_child("static_region");
+    output << boost::format("  %-20s : %s \n") % "XSA Name" % pt_static_region.get<std::string>("vbnv");
+    output << boost::format("  %-20s : %s \n") % "FPGA Name" % pt_static_region.get<std::string>("fpga_name");
+    output << boost::format("  %-20s : %s \n") % "JTAG ID Code" % pt_static_region.get<std::string>("jtag_idcode");
+    
+    boost::property_tree::ptree& pt_board_info = pt_platform.get_child("off_chip_board_info");
+    output << boost::format("  %-20s : %s Bytes\n") % "DDR Size" % pt_board_info.get<std::string>("ddr_size_bytes");
+    output << boost::format("  %-20s : %s \n") % "DDR Count" % pt_board_info.get<std::string>("ddr_count");
+    
+    boost::property_tree::ptree& pt_status = pt_platform.get_child("status");
+    output << boost::format("  %-20s : %s \n") % "Mig Calibrated" % pt_status.get<std::string>("mig_calibrated");
+    output << boost::format("  %-20s : %s \n") % "P2P Status" % pt_status.get<std::string>("p2p_status");
+
+    boost::property_tree::ptree& clocks = pt_platform.get_child("clocks", empty_ptree);
+    for(auto& kc : clocks) {
+      boost::property_tree::ptree& pt_clock = kc.second;
+      output << boost::format("  %-20s : %s MHz\n") % pt_clock.get<std::string>("id") % pt_clock.get<std::string>("freq_mhz");
+    }
+
+    boost::property_tree::ptree& macs = pt_platform.get_child("macs", empty_ptree);
+    for(auto& km : macs) {
+      boost::property_tree::ptree& pt_mac = km.second;
+      output << boost::format("  %-20s : %s\n") % "Mac Address" % pt_mac.get<std::string>("address");

```

Comment:
Similar idea wit the MAC addresses.  For example

```
MAC Addresses
                                : ab:cd:ef:00:01
                                : ab:cd:ef:00:01
```
Or do you have an alternate solution?

Reply:
You are right about repeating the header "Mac addresses" multiple times. As of now I can't think of anything better so I'll implement what you suggest here